lib/editor/result.js

```diff
@@ -0,0 +1,281 @@
+        }
+      }
+    }
+  }
+
+  getMarkerText () {
+    return this.editor.getTextInRange(this.marker.getBufferRange()).trim()
+  }
+
+  // static methods
+
+  static all () {
+    let results = []
+    for (let item of atom.workspace.getPaneItems()) {
+      if (!atom.workspace.isTextEditor(item) || !layers[item.id]) continue
+      layers[item.id].getMarkers().forEach(m => results.push(m.result))
+    }
+    return results
+  }
+
+  static invalidateAll () {
+    for (let result of Result.all()) {
+      delete result.text
+      result.invalidate()
+    }
+  }
+
+  static forLines (ed, start, end, type = 'any') {
+    if (!layers[ed.id]) return
+    return layers[ed.id].findMarkers({intersectsBufferRowRange: [start, end]})
+                        .filter((m) => (type === 'any' || m.result.type === type))
+                        .map((m) => m.result)
+  }
+
+  static removeLines (ed, start, end, type = 'any') {
+    let rs = Result.forLines(ed, start, end, type)
+    if (!rs) return
+    rs.map((r) => r.remove())
+    rs.length > 0
+  }
+
+  static removeAll (ed = atom.workspace.getActiveTextEditor()) {
+    if (!layers[ed.id]) return
+    layers[ed.id].getMarkers().forEach((m) => m.result.remove())
+  }
+
+  static removeCurrent (e) {
+    const ed = atom.workspace.getActiveTextEditor()
+    let done = false
+    if (ed) {
+      for (let sel of ed.getSelections()) {
+        if (Result.removeLines(ed, sel.getHeadBufferPosition().row, sel.getTailBufferPosition().row)) {
+          done = true
+        }
+      }
+    }
+    if (!done) e.abortKeyBinding()
+  }
+
+  static toggleCurrent () {
+    const ed = atom.workspace.getActiveTextEditor()
+    for (const sel of ed.getSelections()) {
+      const rs = Result.forLines(ed, sel.getHeadBufferPosition().row, sel.getTailBufferPosition().row)
+      if (!rs) continue
+      rs.forEach((r) => r.toggle())
+    }
+  }
+
+  // Commands
+
+  static activate () {
+    this.subs = new CompositeDisposable()
+    this.subs.add(atom.commands.add('atom-text-editor:not([mini])',
+      {
+        'inline:clear-current': (e) => Result.removeCurrent(e),
+        'inline-results:clear-all': () => Result.removeAll(),
+        'inline-results:toggle': () => Result.toggleCurrent()
+      }))
+
+    let listener = () => {
+      setTimeout(() => {
+        for (let edid in layers) {
+          let res = layers[edid].getMarkers().map((m) => m.result)
+          res.filter((r) => r.style === 'inline')
+          if (res.length === 0) continue
+          let rect = null
+          fastdom.measure(() => {
+            rect = res[0].editor.editorElement.getBoundingClientRect()
+            res.forEach((r) => r.view.decideUpdateWidth(rect))
+          })
+
+          fastdom.mutate(() => {
+            res.forEach((r) => r.view.updateWidth(rect))
+          })
+        }
+        process.nextTick(() => window.requestAnimationFrame(listener))
+      }, 15*1000/60)
+    }
+    window.requestAnimationFrame(listener)
```

Comment:
I don't think waiting for an animation frame achieves much here if we then wait again. Is it worse if we just loop?

I think this could be optimised by only applying to the active editor. Then you could also have a fast path where it just skips if the width hasn't changed.